File: libdata/src/main/java/com/simprints/libdata/models/firebase/fb_RefusalForm.java
package com.simprints.libdata.models.firebase;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.google.firebase.database.ServerValue;
import com.simprints.libsimprints.RefusalForm;

import java.util.Map;

@SuppressWarnings("WeakerAccess")
public class fb_RefusalForm {
    public String reason;
    public String otherText;
    public String apiKey;
    public String userId;
    public String sessionId;
    public Map<String, String> serverTimestamp;

    public fb_RefusalForm() {
    }

    public fb_RefusalForm(@NonNull RefusalForm refusalForm, @NonNull String apiKey,
                          @NonNull String userId, @NonNull String sessionId) {
        this.reason = nullToEmpty(refusalForm.getReason());
        this.otherText = nullToEmpty(refusalForm.getExtra());
        this.apiKey = apiKey;
        this.userId = userId;
        this.sessionId = sessionId;
        this.serverTimestamp = ServerValue.TIMESTAMP;
    }

    @NonNull
    private String nullToEmpty(@Nullable String value) {
        return value != null ? value : "";
    }
}

File: id/src/main/java/com/simprints/id/data/db/remote/models/fb_RefusalForm.kt
package com.simprints.id.data.db.remote.models

import com.google.firebase.database.ServerValue
import com.simprints.id.domain.refusal_form.IdRefusalForm

class fb_RefusalForm(
    var reason: String,
    var otherText: String,
    var projectId: String,
    var userId: String,
    var sessionId: String,
    var serverTimestamp: Map<String, String>) {

    constructor(refusalForm: IdRefusalForm,
                projectId: String,
                userId: String,
                sessionId: String): this (
        reason = refusalForm.reason ?: "",
        otherText = refusalForm.extra ?: "",
        projectId = projectId,
        userId = userId,
        sessionId = sessionId,
        serverTimestamp = ServerValue.TIMESTAMP
    )
}
